[dynamo] fix keyerror in resume_execution (again)

williamwen42 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #166040
* #166036

Fixes https://github.com/pytorch/pytorch/issues/166176

The error I attempted to fix in https://github.com/pytorch/pytorch/pull/162318 was still appearing internally.

Surprised that this wasn't caught anywhere 😰 

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @Lucaskabela

williamwen42 committed:
[dynamo] fix keyerror in resume_execution (again)

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo] fix keyerror in resume_execution (again)"


The error I attempted to fix in https://github.com/pytorch/pytorch/pull/162318 was still appearing internally.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

Lucaskabela reviewed (approved):
Thanks for continuing to drive to fix this key error - hopefully the additional unit test catch everything and close our gap of internal vs OSS here

williamwen42 committed:
Update on "[dynamo] fix keyerror in resume_execution (again)"


Fixes https://github.com/pytorch/pytorch/issues/166176

The error I attempted to fix in https://github.com/pytorch/pytorch/pull/162318 was still appearing internally.

Surprised that this wasn't caught anywhere 😰 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]
